### PR TITLE
Rewrite using Astro and enhance accordingly

### LINKED ISSUE #69: Rebuild website using modern webframework
As we keep adding more and more pages that all have similar layouts and as such keep re-using certain elements (header, footer, ...) over all pages by just copy-pasting those elements we increase the chance of making mistakes when updating these elements as it becomes easier and easier to miss a page,

Ideally we would use some component based system to define an element once and just re-use that element on each page then.

On the flipside lots of frameworks tend to add massive amounts of JavaScript to create fancy looking components. Preferably we would try to keep the website free from lots of additional JavaScript libraries and just render a static webpage with minimal JS usage.

### LINKED ISSUE #71: Style roadmap page more nicely
Right now it looks super plain
image

Maybe having cards with visuals for each entry as well as some more detailed description would be nice. Happy to see any suggestions brought forward and/or PR'd.

### PR DESCRIPTION
**Pull request by barnabwhy**
All changes are *clearly* objective improvements :3

<details>
<summary>Main page</summary>

(with collapsed installers for demonstration)

</details>

<details>
<summary>Credits</summary>

</details>

Installers page: GONE!!!!!

<details>
<summary>Roadmap</summary>

</details>

<details>
<summary>Servers (unchanged aside from header)</summary>

</details>

closes #69 
closes #71
supersedes #74

### CODE REVIEW

**Comment by GeckoEidechse** [comment]
mf even added TypeScript on the side as well, I love it <3

File: src/pages/roadmap.astro
Comment (GeckoEidechse) [inline src/pages/roadmap.astro:1]: I know this is on a separate page cause it builds off of #74 and so while we are at restructuring the website I think we can also move this onto the main page as well, completely getting rid of the extra page.

Probably between _Installation_ and _Contributor_ section but feel free to put it wherever you see it fit best ^^

**Review by GeckoEidechse — COMMENTED** [review]
Played around with it locally and I like the change a lot :D
Especially the the semi-translucient background on the extra pages and switching all the JS to TypeScript ^^

File: src/pages/index.astro
Comment (GeckoEidechse) [inline src/pages/index.astro:1]: Do `.astro` files default to `TABS` over spaces? At least `Layout.astro` and `index.astro` among others use `TABS` while `404.astro` uses `SPACES` like most files. Should we just quickly set everything to `SPACES` before merging so that it's consistent ^^

**Comment by GeckoEidechse** [comment]
@pg9182 for merging I suppose we just add an Action to build to `github-pages` branch or something and then pull from there on masterserver?

File: src/pages/index.astro
Comment (barnabwhy) [reply src/pages/index.astro:1]: Yeah this might have just been me being dumb, will make it all spaces :)

**Comment by barnabwhy** [comment]
i guess closes #69 and #74

File: src/pages/404.astro
Comment (GeckoEidechse) [inline src/pages/404.astro:35]: There was one tiny tab still hidden here ^^
```suggestion
    body {
```

File: src/pages/roadmap.astro
Comment (GeckoEidechse) [reply src/pages/roadmap.astro:1]: Resolved with 409447c725beeb5959149ee4e2c9a1e625b0a20e

**Review by GeckoEidechse — APPROVED** [review]
Good to merge from my side. Now we just gotta figure out deployment together with @pg9182 ^^

**Comment by GeckoEidechse** [comment]
We should probably also add basic instructions on how to build locally for anyone not familiar with astro but that shouldn't block this PR ^^

**Comment by barnabwhy** [comment]
> We should probably also add basic instructions on how to build locally for anyone not familiar with astro but that shouldn't block this PR ^^

should just be
```sh
$ yarn
$ yarn build
```
and then built files will be in ./dist

**Comment by GeckoEidechse** [comment]
> should just be
> 
> ```shell
> $ yarn
> $ yarn build
> ```
> 
> and then built files will be in ./dist

Exactly.  It's not difficult or anything, it's just that someone who never used NodeJS or the like might not be familiar with this. So just putting the commands somewhere is more than enough.

(Also, `yarn dev` not `yarn build`  for live preview :P)

**Comment by GeckoEidechse** [comment]
Waiting on a response from @pg9182 so that we can merge ^^

**Review by GeckoEidechse — APPROVED** [review]
@pg9182 gave me his blessing to test in prod. It worked using my fork that also adds an Action to deploy and so I'm gonna merge the PR accordingly and the PR the Action to deploy separately

Rendered from https://northstar.tf
image

**Pull request merged**
